src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -1675,6 +1675,72 @@ public SyntaxNode TypeExpression(ITypeSymbol typeSymbol, bool addImport)
         public abstract SyntaxNode NullableTypeExpression(SyntaxNode type);
 
         /// <summary>
+        /// Creates an expression that denotes a tuple type.
+        /// </summary>
+        public SyntaxNode TupleTypeExpression(IEnumerable<SyntaxNode> elements)
+        {
+            if (elements == null)
+            {
+                throw new ArgumentNullException(nameof(elements));
+            }
+            if (elements.Count() <= 1)
+            {
+                throw new ArgumentException("Tuples must have at least two elements.", nameof(elements));
+            }
+
+            return CreateTupleType(elements);
+        }
+
+        protected abstract SyntaxNode CreateTupleType(IEnumerable<SyntaxNode> elements);
+
+        /// <summary>
+        /// Creates an expression that denotes a tuple type.
+        /// </summary>
+        public SyntaxNode TupleTypeExpression(params SyntaxNode[] elements)
+            => TupleTypeExpression((IEnumerable<SyntaxNode>)elements);
+
+        /// <summary>
+        /// Creates an expression that denotes a tuple type.
+        /// </summary>
+        public SyntaxNode TupleTypeExpression(IEnumerable<ITypeSymbol> elementTypes, IEnumerable<string> elementNames = null)
+        {
+            if (elementTypes == null)
+            {
+                throw new ArgumentNullException(nameof(elementTypes));
+            }
+
+            if (elementNames != null)
+            {
+                if (elementNames.Count() != elementTypes.Count())
+                {
+                    throw new ArgumentException("The number of element names must match the cardinality of the tuple.", nameof(elementNames));
+                }
+                return TupleTypeExpression(elementTypes.Zip(elementNames, AsTupleElement));
+            }
+
+            return TupleTypeExpression(elementTypes.Select(AsTupleElement));
+        }
+
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+        /// <summary>
+        /// Creates an expression that denotes a tuple element.
+        /// </summary>
+        public abstract SyntaxNode TupleElementExpression(SyntaxNode type, string name = null);
+
+        /// <summary>
+        /// Creates an expression that denotes a tuple element.
+        /// </summary>
+        public SyntaxNode TupleElementExpression(ITypeSymbol type, string name = null)
+            => TupleElementExpression(TypeExpression(type), name);
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+
+        private SyntaxNode AsTupleElement(ITypeSymbol type)
+            => TupleElementExpression(type, name: null);
+
+        private SyntaxNode AsTupleElement(ITypeSymbol type, string name)
+            => TupleElementExpression(type, name);
```

Comment:
these as methods seem to be called only from one place.  Inline.   (they'll also be cheaper as we will cache the delegates insstead of creating them each time).